scalableminds/webknossos
# Silhouette combined authenticator

### Mailable description of changes:
 - users can now request a token for authentication via the X-Auth-Token mechanism

### Changes detailed
 - implemented a combinedAuthenticator and combinedAuthenticatorService
 - combines cookie and token behavior
 - the default is cookie
 - the token is stored in the request header (in the "X-Auth-Token"-field)
 - tokens are saved in the DB (in the combinedAuthenticator collection)
 - cookies get (un)serialized and not saved in the DB. (Alternatively they could also be save in the DB) 

### Steps to test:
- try to log in 
- try to get a token (/api/getToken) (you need be be logged in)
- try to get another token (the old token should be deleted)
- try to authenticate yourself with the token
- delete token (/api/deleteToken)
- logout
- try autologin in dev mode

------
